pkg/runtime/service_controller.go

```diff
@@ -227,7 +227,7 @@ func (c *serviceController) BindControllerManager(mgr ctrlrt.Manager, cfg ackcfg
 	// The caches are only used for cross account resource management. If the
 	// controller is not configured to watch multiple namespaces, then we don't
 	// need to run the caches.
-	if len(namespaces) == 0 || len(namespaces) >= 2 {
+	if cfg.EnableCARM {
```

Comment:
i'm not opposed to this, but if you want to allow users to enable CARM even they're on single namespace mode, should we update the comment above?